content/en/blog/_posts/2021-12-15-pod-security-admission-psa-beta.md

```diff
@@ -0,0 +1,620 @@
+
+```shell
+# kube-test-privileged enforces a "privileged" security policy and warns / audits on baseline
+kubectl label --overwrite ns kube-test-privileged \
+  pod-security.kubernetes.io/enforce=privileged \
+  pod-security.kubernetes.io/warn=baseline \
+  pod-security.kubernetes.io/audit=baseline
+
+# kube-test-baseline enforces a "baseline" security policy and warns / audits on restricted
+kubectl label --overwrite ns kube-test-baseline \
+  pod-security.kubernetes.io/enforce=baseline \
+  pod-security.kubernetes.io/warn=restricted \
+  pod-security.kubernetes.io/audit=restricted
+
+# kube-test-restricted enforces a "restricted" security policy and audits on restricted
+kubectl label --overwrite ns kube-test-restricted \
+  pod-security.kubernetes.io/enforce=restricted \
+  pod-security.kubernetes.io/audit=restricted
+```
+
+### Deploy demo workloads
+
+Each workload represents a higher level of security that would not pass the profile that comes after it.
+
+**Privileged level and workload**
+
+For the privileged pod, allow privilege escalation. This allows the process inside a container to gain new processes and can be dangerous if untrusted ([more info](https://kubernetes.io/docs/concepts/security/pod-security-standards/#privileged)).
+
+First, let's try to deploy it in the restricted namespace:
+
+```shell
+cat <<EOF | kubectl -n kube-test-restricted apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx-privileged
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    securityContext:
+      allowPrivilegeEscalation: true
+EOF
+```
+
+The output is similar to this:
+
+```shell
+Error from server (Forbidden): error when creating "STDIN": pods "nginx-privileged" is forbidden: violates PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "nginx" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "nginx" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "nginx" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "nginx" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
+```
+
+Now let's try the `kube-test-privileged` namespace
+
+```shell
+cat <<EOF | kubectl -n kube-test-privileged apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx-privileged
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+    securityContext:
+      allowPrivilegeEscalation: true
+EOF
+```
+
+The output is similar to this:
+
+```shell
+pod/nginx-privileged created
+```
+
+We can run `kubectl -n kube-test-privileged get pods` to verify it is running.
+
+**Baseline level and workload**
+
+The baseline pod demonstrates sensible defaults but does not allow privilege escalation, so the pod above would fail here too ([more info](https://kubernetes.io/docs/concepts/security/pod-security-standards/#baseline)). Baseline accepts null / no values for unset keys.
```

Comment:
"more info" is not ideal link text because of how SEO and screenreaders work. I suggest moving the link to the word "baseline" at the start of the sentence, and removing the "more info".